### PR TITLE
Convert class type from model to record.

### PR DESCRIPTION
**Pull request by hyumo**
see issue #17 for detailed information

### CODE REVIEW

**Comment by tbeu** [comment]
@HansOlsson May I kindly ask for your review, since I have no strong opinion if model or record is to be preferred.

**Comment by hyumo** [comment]
Thanks @tbeu 

So what I was trying to do (or there might be some better ways) was to read `Integer` stored in XML as `enumeration`. 

For example, for `Modelica.Blocks.Types.Init`, I have to use `1` for  `NoInit`, `2` for `SteadyState`, etc... 

I then create a function for this 
```
function readXmlInitType
   input String xmlpath;
   input ExternData.XMLFile xml;
   output Modelica.Blocks.Types.Init init_type;
protected
   Integer init_enum;
algorithm
  init_enum := xml.getInteger(xmlpath);
  if init_enum == 1 then
     init_type := Modelica.Blocks.Types.Init.NoInite
  elseif 
     ...
  endif;
end function
```

This is where I prefer passing the XMLFile as an argument.

File: ExternData/package.mo [hunk 1]
```diff
@@ -57,7 +57,7 @@ package ExternData "Library for data I/O of CSV, INI, JSON, MATLAB MAT, Excel XL
     final parameter Types.ExternCSVFile csv=Types.ExternCSVFile(fileName, delimiter, quotation, verboseRead) "External INI file object";
     final function getRealArray2D = Functions.CSV.getRealArray2D(final csv=csv) "Get 2D Real values from CSV file" annotation(Documentation(info="<html></html>"));
     annotation(
-      Documentation(info="<html><p>Model that wraps the external object <a href=\"modelica://ExternData.Types.ExternCSVFile\">ExternCSVFile</a> and the <a href=\"modelica://ExternData.Functions.CSV\">CSV</a> read function for data access of <a href=\"https://en.wikipedia.org/wiki/Comma-separated_values\">CSV</a> files.</p><p>See <a href=\"modelica://ExternData.Examples.CSVTest\">Examples.CSVTest</a> for an example.</p></html>"),
+      Documentation(info="<html><p>record that wraps the external object <a href=\"modelica://ExternData.Types.ExternCSVFile\">ExternCSVFile</a> and the <a href=\"modelica://ExternData.Functions.CSV\">CSV</a> read function for data access of <a href=\"https://en.wikipedia.org/wiki/Comma-separated_values\">CSV</a> files.</p><p>See <a href=\"modelica://ExternData.Examples.CSVTest\">Examples.CSVTest</a> for an example.</p></html>"),
```
Comment (tbeu) [inline ExternData/package.mo:60]: Could you please use captial `R` here (and all other occurences): Record

**Comment by hyumo** [comment]
Sorry that I deleted my fork... I've made a new pull request.

**Pull request closed without merging**